user_manager/api/v1/views.py

```diff
@@ -0,0 +1,419 @@
+        queryset(QuerySet): UserManagerRole queryset
+        manager_id(str): username or email address of manager
+    Returns:
+        queryset filtered by manager
+    """
+
+    if manager_id is None:
+        return queryset
+    elif '@' in manager_id:
+        return queryset.filter(
+            Q(manager_user__email=manager_id) | Q(unregistered_manager_email=manager_id),
+        )
+    else:
+        return queryset.filter(
+            manager_user__username=manager_id,
+        )
+
+
+def _filter_by_user_id(queryset, user_id):
+    """
+    Filter ``queryset`` by ``user_id``, where ``user_id`` can be a username or email address.
+
+    Args:
+        queryset(QuerySet): UserManagerRole queryset
+        user_id(str): username or email address of user
+    Returns:
+        queryset filtered by user
+    """
+
+    if user_id is None:
+        return queryset
+    elif '@' in user_id:
+        return queryset.filter(user__email=user_id)
+    else:
+        return queryset.filter(user__username=user_id)
+
+
+class ManagerViewMixin(object):
+    """
+    Provide common functionality for all manager views.
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    @property
+    def authentication_classes(self):  # pragma: no cover
+        """
+        Allow users authenticated via OAuth2 or normal session authentication.
+        """
+        from openedx.core.lib.api import authentication  # pylint: disable=import-error
+        return [
+            authentication.OAuth2AuthenticationAllowInactiveUser,
+            authentication.SessionAuthenticationAllowInactiveUser,
+        ]
+
+
+class ManagerListView(ManagerViewMixin, ListAPIView):
+    """
+        **Use Case**
+
+            * Get a list of all users that are managers for other users
+
+        **Example Request**
+
+            GET /api/user_manager/v1/managers/
+
+        **GET Parameters**
+
+            None
+
+        **GET Response Values**
+
+            If the request for information about the managers is successful, an HTTP 200 "OK"
+            response is returned with a collection of managers.
+
+            The HTTP 200 response has the following values.
+
+            * count: The number of managers in a course.
+
+            * next: The URI to the next page of results.
+
+            * previous: The URI to the previous page of results.
+
+            * num_pages: The number of pages.
+
+            * results: a list of manager users:
+
+                * id: The user id for a manager user, or null if manager doesn't have an
+                    account yet.
+
+                * email: Email address of manager.
+
+        **Example GET Response**
+
+            {
+                "count": 99,
+                "next": "https://courses.org/api/user_manager/v1/managers/?page=2",
+                "previous": null,
+                "results": {
+                    {
+                        "email": "staff@example.com",
+                        "id": 9
```

Comment:
REST APIs should return usernames, not IDs.

https://openedx.atlassian.net/wiki/spaces/OpenDev/pages/18350757/edX+REST+API+Conventions#edXRESTAPIConventions-2.IdentifyingResources

It would probably be nice to include the user's full names here too, but not necessary.